eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/bag/SynchronizedPutHashBagMultimap.java

```diff
@@ -161,6 +161,11 @@ public void value(K key, MutableBag<V> bag)
         return this.collectValues(function, HashBagMultimap.<K, V2>newMultimap());
     }
 
+    public MutableBagMultimap<K, V> asSynchronized()
+    {
+        return SynchronizedMutableBagMultimap.of(this);
```

Comment:
This is an interesting case. `SynchronizedPutHashBagMultimap.asSynchronized()` returning a wrapper with a second lock.